packages/web/components/complex/all-pools-table-v2.tsx

```diff
@@ -90,23 +146,24 @@ export const AllPoolsTable: FunctionComponent<{
   } = api.edge.pools.getMarketIncentivePools.useInfiniteQuery(
     {
       limit: 100,
-      search: searchQuery,
-      types: poolTypesFilter
-        ? poolTypesFilter.concat("cosmwasm" as PoolTypeFilter)
+      search: filters.searchQuery
+        ? {
+            query: filters.searchQuery,
+          }
         : undefined,
-      incentiveTypes: poolIncentivesFilter,
+      types: [...filters.poolTypesFilter, "cosmwasm"],
+      incentiveTypes: filters.poolIncentivesFilter,
       sort: sortKey
         ? {
             keyPath: sortKey,
-            direction: sortDirection,
+            direction: sortParams.allPoolsSortDir,
           }
         : undefined,
       minLiquidityUsd: 1_000,
     },
     {
       getNextPageParam: (lastPage) => lastPage.nextCursor,
       initialCursor: 0,
-      enabled: isQueryParamsReady,
```

Comment:
I used this to prevent queries from being sent before query param was loaded in future renders. It seems nuqs somehow loads query params on first render? As I don't see any queries with unloaded inputs, which is very nice.

Reply:
Yes, because it actually works differently, it builds an object with default values, if we change the default values then it puts them in the url.
This avoids having to wait for the router to be ready, plus if you don't change the filters in the url there are no "redundant" strings